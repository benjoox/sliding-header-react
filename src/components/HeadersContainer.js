import React, { useState, useRef, useEffect } from 'react'
import { useScroll, useResize } from './CustomHooks'
import HeaderPrimary from './HeaderPrimary'
import HeaderSecondary from './HeaderSecondary'
import '../index.css'

export default function HeadersContainer() {
  const el = useRef(null)
  const [collapsed, setCollapsed] = useState(false)
  const [innerWidth, setInnerWidth] = useState(1)
  const [width, setWidth] = useState(0)
  useScroll((e) => {
    if (window.scrollY > 50) setCollapsed(window.scrollY > 50)
  })
  useResize((e) => {
    setInnerWidth(window.innerWidth)
  })

  useEffect(() => {
    if (el.current) {
      setWidth(el.current.offsetWidth)
    }
  }, [el, innerWidth])
  return (
    <div className={`container ${collapsed ? 'collapsed' : ''}`}>
      <header className={collapsed ? 'collapsed' : ''}>
        <HeaderPrimary width={width} collapsed={collapsed} />
        <HeaderSecondary width={width} collapsed={collapsed} />
      </header>
    </div>
  )
}
