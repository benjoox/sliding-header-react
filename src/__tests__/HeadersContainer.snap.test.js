import React from 'react'
import renderer from 'react-test-renderer'
import HeadersContainer from '../components/HeadersContainer'

it('HeadersContainer  renders correctly', () => {
  const tree = renderer.create(<HeadersContainer />).toJSON()
  expect(tree).toMatchSnapshot()
})
