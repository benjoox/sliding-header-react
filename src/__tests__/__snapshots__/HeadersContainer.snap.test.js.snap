// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadersContainer  renders correctly 1`] = `
<div
  className="container "
>
  <header
    className=""
  >
    <div
      className="primary"
    >
      <section>
        <div
          className="vertical-align"
        >
          <svg>
            forum-dots.svg
          </svg>
        </div>
        <div
          className="vertical-align text-logo "
        >
          <svg>
            forum-logo-text.svg
          </svg>
        </div>
      </section>
      <section>
        <div
          className="search-box-container"
        >
          <div
            className="search-box "
          >
            <div
              className="search-icon "
            >
              <svg>
                search.svg
              </svg>
            </div>
            <input
              className="search-box-input"
              placeholder="Search"
              type="text"
            />
          </div>
        </div>
        <button
          className="registeration-btn"
        >
          <div>
            <svg>
              profile.svg
            </svg>
          </div>
          <p>
            Register
          </p>
        </button>
      </section>
    </div>
    <div
      className="secondary "
    >
      <ul>
        <li
          className="dropdown-item"
        >
          All A-Z
          <span>
            <svg>
              arrow-down.svg
            </svg>
          </span>
        </li>
        <li
          className="topic-item"
        >
          <a
            href="#"
          >
            Brexit
          </a>
        </li>
        <li
          className="topic-item"
        >
          <a
            href="#"
          >
            Climate
          </a>
        </li>
        <li
          className="topic-item"
        >
          <a
            href="#"
          >
            Copyright
          </a>
        </li>
        <li
          className="topic-item"
        >
          <a
            href="#"
          >
            Migration
          </a>
        </li>
        <li
          className="topic-item"
        >
          <a
            href="#"
          >
            Deb on hover
          </a>
        </li>
        <li
          className="topic-item"
        >
          <a
            href="#"
          >
            Debate X
          </a>
        </li>
        <li
          className="topic-item"
        >
          <a
            href="#"
          >
            Debate Y
          </a>
        </li>
        <li
          className="topic-item"
        >
          <a
            href="#"
          >
            Debate L
          </a>
        </li>
        <li
          className="topic-item"
        >
          <a
            href="#"
          >
            Debate S
          </a>
        </li>
      </ul>
    </div>
  </header>
</div>
`;
